kaitai/kaitaistream.cpp

```diff
@@ -112,30 +112,30 @@ int8_t kaitai::kstream::read_s1() {
 // ........................................................................
 
 int16_t kaitai::kstream::read_s2be() {
-    int16_t t;
+    uint16_t t;
     m_io->read(reinterpret_cast<char *>(&t), 2);
 #if __BYTE_ORDER == __LITTLE_ENDIAN
     t = bswap_16(t);
 #endif
-    return t;
+    return int16_t(t);
```

Comment:
Can you explain what's the purpose of this change?

Reply:
`bswap_{16,32,64}` takes unsigned int, the function returns signed int which causes a sign conversion warning.